app/models/shelter.rb

```diff
@@ -16,6 +18,14 @@ def self.order_by_number_of_pets
       .order("pets_count DESC")
   end
 
+  def self.order_by_reverse_alphabetical
+    find_by_sql("SELECT * FROM shelters ORDER BY name DESC")
+  end
+
+  def self.pending_apps
+    joins(:applications).where("status = 'Pending'")
```

Comment:
With the associations you have set up, `applications` is now an attribute on shelters so you can skip the joins method here and just start with `applications.where`